src/AElf.Runtime.CSharp.Core/Validators/Method/MultiDimArrayValidator.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace AElf.Runtime.CSharp.Validators.Method
+{
+    public class MultiDimArrayValidator : IValidator<MethodDefinition>
+    {
+        public IEnumerable<ValidationResult> Validate(MethodDefinition method)
+        {
+            if (!method.HasBody)
+                return Enumerable.Empty<ValidationResult>();
+            
+            var errors = new List<FloatValidationResult>();
+            
+            foreach (var instruction in method.Body.Instructions)
+            {
+                if (instruction.OpCode.Code == Code.Newobj)
+                {
+                    var methodRef = (MethodReference) instruction.Operand;
+
+                    if (methodRef.DeclaringType.IsArray && ((ArrayType) methodRef.DeclaringType).Dimensions.Count > 1)
+                    {
+                        return new List<ValidationResult>
+                        {
+                            new MultiDimArrayValidationResult(
```

Comment:
This is inconsistent with adding all results in output like in UnsafeMathValidator

Reply:
Done